common/src/main/java/com/microsoft/identity/common/internal/ui/AndroidAuthorizationStrategyFactory.java

```diff
@@ -54,69 +47,72 @@
 @SuppressWarnings(WarningType.rawtype_warning)
 @Builder
 @Accessors(prefix = "m")
-public class AndroidAuthorizationStrategyFactory implements IAuthorizationStrategyFactory{
+public class AndroidAuthorizationStrategyFactory implements IAuthorizationStrategyFactory<IAuthorizationStrategy> {
     private static final String TAG = AndroidAuthorizationStrategyFactory.class.getSimpleName();
 
     private final Context mContext;
     private final Activity mActivity;
     private final Fragment mFragment;
 
+    /**
+     * Get the authorization strategy.
+     *
+     * @param authorizationAgent The authorization agent provided by the caller.
+     * @param browser            The browser to use for authorization.
+     * @param isBrokerRequest   True if the request is from broker.
+     * @return The authorization strategy.
+     */
     @Override
+    @NonNull
     public IAuthorizationStrategy getAuthorizationStrategy(
-            @NonNull final InteractiveTokenCommandParameters parameters) {
+            @NonNull final AuthorizationAgent authorizationAgent,
+            @Nullable final Browser browser,
+            final boolean isBrokerRequest) {
         final String methodTag = TAG + ":getAuthorizationStrategy";
-        //Valid if available browser installed. Will fallback to embedded webView if no browser available.
 
-        if (parameters.getAuthorizationAgent() == AuthorizationAgent.WEBVIEW) {
-            Logger.info(methodTag, "Use webView for authorization.");
+        // Use embedded webView if no browser available or authorization agent is webView
+        if (authorizationAgent == AuthorizationAgent.WEBVIEW || browser == null) {
```

Comment:
If someone passes authorization agent as BROWSER but passes null browser, then what strategy is picked?

Reply:
embedded web view authorization strategy

Add Javadoc:
* Get the authorization strategy based on the authorization agent and browser.
* If the authorization agent is WEBVIEW or the browser is null,
* return the embedded web view authorization strategy.
* Otherwise, return the browser authorization strategy.